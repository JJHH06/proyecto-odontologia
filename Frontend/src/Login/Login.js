/* eslint-disable prefer-template */
import React, { useState } from 'react';

// objeto de la calculadora
function Login() {

  return (
    <div>
        <h1>Login</h1>
    </div>
  );
}

export default Login;
