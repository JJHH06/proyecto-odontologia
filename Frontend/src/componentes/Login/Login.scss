@import url('https://fonts.googleapis.com/css2?family=Poppins&display=swap');

.login{
    font-family: 'Poppins', sans-serif !important;
    background: url('https://gacetadental.com/wp-content/uploads/2019/12/Clinica-dental-1-1280x720.jpg') no-repeat center center fixed;
    background-size: cover;
    width: 100%;
    height: 100%;
}

.content{
    margin: 10% !important;
    background-color: rgba($color: #ffffff, $alpha: .8);
    padding: 4rem 1rem 4rem 1rem;
    box-shadow: 0 0 10px 10px rgba(0, 0, 0, .5);
    border-radius: 10px;
}

@media (max-width: 400px) {
    .content{
        margin: 8% !important;
        
    }
}

.posicion{
    margin-top: 50px;
}

.signin-text{
    font-style: normal;
    font-weight: 600 !important;

}

.form-control{
    display: block;
    width: 100%;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1.5;
    border-color: rgb(74, 193, 224) !important;
    border-style: solid !important;
    border-width: 0 0 1px 0 !important;
    padding: 0px !important;
    color: #495057;
    height: auto;
    border-radius: 0;
    background-color: #fff;
    background-clip: padding-box;

}

.espacio{
    margin-top: 20px;
}

.form-control:focus{
    color: #495057;
    background-color: #fff;
    border-color: #fff;
    outline: 0;
    box-shadow: none;
}

.form-group.form-check{
    margin-top: 20px;
}

.container .row.content input[type='submit']{
    border: 1px solid rgb(74, 193, 224);
    outline: none;
    margin-top: 20px;
    
}

.btn.btn-class.boton{
    border: 1px solid rgb(74, 193, 224);
    outline: none;
    margin-top: 20px;
    margin-left: 20px;
}

.btn-class{
    border-style: solid;
    border-color: rgb(74, 193, 224);
    color: rgb(74, 193, 224);
}

.btn-class:hover{
    background-color: rgb(74, 193, 224);
    color: rgb(74, 193, 224);
}

