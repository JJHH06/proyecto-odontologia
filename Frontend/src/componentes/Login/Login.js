/* eslint-disable prefer-template */
import React, { useState } from "react";
import "bootstrap";
import "./Login.scss";
import Logotipo from "../../assets/FULL_COLOR.png";
import PropTypes from "prop-types";
import axios from "axios";
import { useHistory } from 'react-router-dom'

// const axios = require('axios');

// function loginConnection (email, pass){
//   var data = JSON.stringify({
//     "id_attemp": email,
//     "pass_attemp": pass
//   });
//   let respuesta = {}

//   var config = {
//     method: 'post',
//     url: 'http://198.211.103.50:5000/api/login/validate_login',
//     headers: {
//       'Content-Type': 'application/json'
//     },
//     data : data
//   };

//   axios(config)
//   .then(function (response) {
//     respuesta = JSON.stringify(response.data)
//     console.log(JSON.stringify(response.data).validate_login);
//     //return JSON.stringify(JSON.stringify(response.data).validate_login)

//   })
//   .catch(function (error) {
//     console.log(error);
//   });
// }

//loginConnection("XD","XDDDD")

const getCurrentUser = async (token, id) => {
  
  const data = JSON.stringify({
    id_empleado: id,
  });

  let config = {
    method: "post",
    url: "http://198.211.103.50:5000/api/empleado/getEmpleado",
    headers: {
      Authorization: "Bearer  " + token,
      "Content-Type": "application/json",
    },
    data: data,
  };

  return await axios(config)
    .then(function (response) {
      return response.data.result;
    })
    .catch(function (error) {
      console.log(error);
    });
};

function Login({ setToken, setCurrentUser }) {
  const [username, setUserName] = useState();
  const [password, setPassword] = useState();
  const history = useHistory();
  const handleSubmit = async (e) => {
    e.preventDefault();
    const token = { token: "token123" };

    const data = JSON.stringify({
      id_attemp: username,
      pass_attemp: password,
    });

    let config = {
      method: "post",
      url: "http://198.211.103.50:5000/api/login/validate_login",
      headers: {
        "Content-Type": "application/json",
      },
      data: data,
    };
    return await axios(config)
      .then(async (response) => {
        if (response.data.result.validate_login) {
          const usuario =await getCurrentUser(response.data.result.token, username)
          setCurrentUser(usuario);
          sessionStorage.setItem('currentUser', usuario);
          history.push("/home")
          setToken({ token: response.data.result.token });
          
          return
          //
        }
        else{
          alert("Usuario o contrase√±a incorrectos");
        }
      })
      .catch(function (error) {
        console.log(error);
      });
  };
  return (
    <div data-testid="login-component" className="login">
      <div className="container">
        <div className="row content">
          <div className="col-md-6 mb-3">
            <img src={Logotipo} className="img-fluid posicion" alt="imagen" />
          </div>
          <div className="col-md-6">
            <h3 className="signin-text mb-3">Inicio de Sesion</h3>
            <form onSubmit={handleSubmit}>
              <div className="form-group">
                <label htmlFor="text">Email</label>
                <input
                  type="text"
                  name="text"
                  className="form-control"
                  onChange={(e) => setUserName(e.target.value)}
                />
              </div>
              <div data-testid="login-form" className="form-group">
                <label htmlFor="password" className="espacio">
                  Password
                </label>
                <input
                  type="password"
                  name="password"
                  className="form-control"
                  onChange={(e) => setPassword(e.target.value)}
                />
              </div>
              <div className="form-group form-check">
                <input
                  type="checkbox"
                  name="checkbox"
                  className="form-check-input"
                />
                <label className="form-check-label" htmlFor="checkbox">
                  Remember Me
                </label>
              </div>
              <input
                type="submit"
                name="submit"
                value="Iniciar sesion"
                className="btn btn-class"
              />
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default Login;

Login.propTypes = {
  setToken: PropTypes.func.isRequired,
};

/*
        <input type='text' name='' placeholder='Username' />
        <input type='password' name='' placeholder= 'Password' />
        <input type='submit' name='' value= 'Login' />
*/
