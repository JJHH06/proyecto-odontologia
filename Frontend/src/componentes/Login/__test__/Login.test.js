import React, {useState}  from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {render, cleanup, screen} from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";
import Login from '../Login';

//Mocks token validation
function setToken(token){}

afterEach(cleanup);
it("Renders without crashing",()=>{
    
    const div = document.createElement("div");
    ReactDOM.render(<Login setToken={setToken} />, div);
    ReactDOM.unmountComponentAtNode(div);
});

it("Renders Login Correctly",() =>{
    
    const {getByTestId}= render(<Login setToken={setToken} />)
    expect(getByTestId('login-component')).toHaveTextContent("Inicio de Sesion")
});

it('Login is possible', async () => {
    
    render(<Login setToken={setToken} />);
    expect(screen.getByTestId('login-form')).toBeInTheDocument();
});

it("Matches snapshot",() =>{
    
    const tree = renderer.create(<Login setToken={setToken} />).toJSON();
    expect(tree).toMatchSnapshot();
})
