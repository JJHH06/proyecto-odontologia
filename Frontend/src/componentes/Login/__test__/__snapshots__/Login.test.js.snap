// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<div
  className="login"
  data-testid="login-component"
>
  <div
    className="container"
  >
    <div
      className="row content"
    >
      <div
        className="col-md-6 mb-3"
      >
        <img
          alt="imagen"
          className="img-fluid posicion"
          src="FULL_COLOR.png"
        />
      </div>
      <div
        className="col-md-6"
      >
        <h3
          className="signin-text mb-3"
        >
          Inicio de Sesion
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              htmlFor="text"
            >
              Email
            </label>
            <input
              className="form-control"
              name="text"
              onChange={[Function]}
              type="text"
            />
          </div>
          <div
            className="form-group"
            data-testid="login-form"
          >
            <label
              className="espacio"
              htmlFor="password"
            >
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              type="password"
            />
          </div>
          <div
            className="form-group form-check"
          >
            <input
              className="form-check-input"
              name="checkbox"
              type="checkbox"
            />
            <label
              className="form-check-label"
              htmlFor="checkbox"
            >
              Remember Me
            </label>
          </div>
          <input
            className="btn btn-class"
            name="submit"
            type="submit"
            value="Iniciar sesion"
          />
        </form>
      </div>
    </div>
  </div>
</div>
`;
